swagger_spec_compatibility/walkers/additional_properties.py

```diff
@@ -39,6 +40,45 @@ def fix_parameter_path(self, path, original_path):
         )
 
 
+CycleSentinel = collections.namedtuple('CycleSentinel', ('path',))
+
+
+def _cycle_safe_dict(dict_with_cycles):
+    # type: (typing.Dict[typing.Any, typing.Any]) -> typing.Dict[typing.Any, typing.Any]
```

Comment:
The dictionary should still be a representation of JSON, so key is `str` :)
```suggestion
    # type: (typing.Dict[str, typing.Any]) -> typing.Dict[str, typing.Any]
```

Reply:
This is true for Swagger specs but not generally for all dicts. Since this function can work with any dict keys I didn't see a need to do this.